src/app/libp2p_helper/src/codanet.go

```diff
@@ -118,9 +137,51 @@ func (cm *CodaConnectionManager) ClosedStream(net network.Network, stream networ
 	cm.p2pManager.Notifee().ClosedStream(net, stream)
 }
 func (cm *CodaConnectionManager) Connected(net network.Network, c network.Conn) {
+	logger.Debugf("%s connected to %s", c.LocalPeer(), c.RemotePeer())
 	cm.OnConnect(net, c)
 	cm.p2pManager.Notifee().Connected(net, c)
+
+	if len(net.Peers()) <= cm.maxConnections {
+		return
+	}
+	logger.Debugf("node=%s disconnecting from peer=%s; max peers=%d peercount=%d", c.LocalPeer(), c.RemotePeer(), cm.maxConnections, len(net.Peers()))
+
+	// select random subset of our peers to send over, then disconnect
+	if cm.getRandomPeers == nil {
+		logger.Error("getRandomPeers function not set")
+		return
+	}
+
+	peers := cm.getRandomPeers(numPeersToExchange, c.RemotePeer())
+	bz, err := json.Marshal(peers)
+	if err != nil {
+		logger.Error("failed to marshal peers", err)
+		return
+	}
+
+	stream, err := cm.host.NewStream(cm.ctx, c.RemotePeer(), pxProtocolID)
+	if err != nil {
+		logger.Error("failed to open stream", err)
+		return
+	}
+
+	n, err := stream.Write(bz)
+	if err != nil {
+		logger.Error("failed to write to stream", err)
+		return		
+	} else if n != len(bz) {
+		logger.Error("failed to write all data to stream")
+		return		
+	}
+
+	logger.Debugf("wrote peers to stream %s", stream.Protocol())
+	go func() {
+		// small delay to allow for remote peer to read from stream
+		time.Sleep(time.Millisecond * 300)
+		_ = c.Close()
```

Comment:
Do you need to close the stream explicitly?

Reply:
closing the connection implicity closes all streams associated with it